swarm/swarm.go

```diff
@@ -177,8 +177,9 @@ func NewSwarm(config *api.Config, mockStore *mock.NodeStore) (self *Swarm, err e
 	}
 	self.streamer = stream.NewRegistry(nodeID, delivery, self.netStore, stateStore, &stream.RegistryOptions{
 		SkipCheck:       config.DeliverySkipCheck,
-		DoSync:          config.SyncEnabled,
+		DoSync:          config.SyncEnabled && !config.LightNodeEnabled,
```

Comment:
since `LightNodeEnabled` might have several implications about toggling of different flags and configuration parameters, I'd rather pull the check for `config.LightNodeEnabled` before this code block is executed and have a:
```
if config.LightNodeEnabled {
 config.SyncEnabled = false
 config.DoServeRetrieve = false
}
```
As an override section.
The flag suggests a general mode of operation and the combination of this into boolean logic is confusing and would make it harder to find problems in the configuration.

Reply:
Fixed

Reply:
ok 👍

Reply:
Regarding `DoServeRetrieve`: it is not coming from the configuration, but it rather it's a part of the RegistryOptions when setting up a new registry. Basically this is the other thing what a light node does differently besides turning off syncing: with this option it does not serve Retrieve requests. 

From the existing code to me it seemed desirable to let these things separately (e.g. DoSync and DoServeRetrieve) and that's why it became part of the RegistryOptions.

I updated the RegistryOption initializing logic according to your recommendation. Please review again :)

Reply:
i did - looks better. can be merged at this point AFAIC